hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -698,7 +686,59 @@ public String getPayloadClass() {
    * Read the payload class for HoodieRecords from the table properties.
    */
   public String getRecordMergerStrategy() {
-    return getStringOrDefault(RECORD_MERGER_STRATEGY);
+    return getString(RECORD_MERGER_STRATEGY);
+  }
+
+  public static Triple<RecordMergeMode, String, String> inferCorrectMergingBehavior(RecordMergeMode recordMergeMode, String payloadClassName, String recordMergerStrategy) {
+    RecordMergeMode inferRecordMergeMode;
+    String inferPayloadClassName;
+    String inferRecordMergerStrategy;
+
+    if (isNullOrEmpty(payloadClassName)) {
+      if (isNullOrEmpty(recordMergerStrategy)) {
+        checkArgument(recordMergeMode != RecordMergeMode.CUSTOM, "Custom merge mode should only be used if you set a merge strategy");
```

Comment:
If would be helpful to add a line of docs to explain each scenario, e.g., `// No payload class name, no record merge strategy`.